tools/importer-rest-api-specs/components/dataapigeneratorjson/transforms/sdk_model.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package transforms
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/pandora/tools/sdk/dataapimodels"
+	"github.com/hashicorp/pandora/tools/sdk/resourcemanager"
+)
+
+func MapSDKModelToRepository(modelName string, model resourcemanager.ModelDetails, parentModel *resourcemanager.ModelDetails, knownConstants map[string]resourcemanager.ConstantDetails, knownModels map[string]resourcemanager.ModelDetails) (*dataapimodels.Model, error) {
+	if len(model.Fields) == 0 {
+		return nil, fmt.Errorf("the model %q has no fields", modelName)
+	}
+
+	fields, err := mapSDKFieldsForModel(model, parentModel, knownConstants, knownModels)
+	if err != nil {
+		return nil, fmt.Errorf("mapping fields for model %q: %+v", modelName, err)
+	}
+
+	dataApiModel := dataapimodels.Model{
+		Name:   modelName,
+		Fields: *fields,
+	}
+
+	// NOTE: `Parent` types don't get a `TypeHintIn` or `TypeHintValue`
+	// meaning that only
```

Comment:
is this comment missing a bit 🤔 ?

Reply:
It is (it's a lift and shift) - but this is actually fixed in one of the follow up PR's so I'm gonna leave this for now